vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java

```diff
@@ -96,9 +96,20 @@
     public void initVariableUsingTemplate(Set<VariableScope> scopes, String variableName, String templatePath,
             ExamplesTable templateParameters) throws IOException, TemplateException
     {
-        Map<String, ?> dataModel = MapUtils.convertExamplesTableToMap(templateParameters);
-        String value = freemarkerProcessor.process(templatePath, Map.of("parameters", dataModel),
-                StandardCharsets.UTF_8);
+        Map<String, List<String>> dataModel = MapUtils.convertExamplesTableToMap(templateParameters);
+        String parametersKey = "parameters";
+        Set<String> keySet = dataModel.keySet();
+        Map<String, Object> valuesMap;
+        if (keySet.size() == 1 && dataModel.entrySet().iterator().next().getValue().size() == 1)
+        {
+            String key = keySet.stream().findFirst().get();
+            valuesMap = Map.of(parametersKey, dataModel, key, dataModel.get(key).get(0));
+        }
+        else
+        {
+            valuesMap = Map.of(parametersKey, dataModel);
+        }
+        String value = freemarkerProcessor.process(templatePath, valuesMap, StandardCharsets.UTF_8);
```

Comment:
```java
Map<String, List<String>> dataModel = MapUtils.convertExamplesTableToMap(templateParameters);
String parametersKey = "parameters";
Map<String, Object> parameters = new HashMap<>();
parameters.put(parametersKey, dataModel);
if (!dataModel.contains(parametersKey))
{
    parameters.putAll(dataModel);
}
String value = freemarkerProcessor.process(templatePath, parameters, StandardCharsets.UTF_8);
```
?

Reply:
@valfirst  this implementation requires  refer to the variable as key[0] (using index)

Reply:
@EDbarvinsky that's expected